contracts/CandidateHub.sol

```diff
@@ -86,16 +87,19 @@ contract CandidateHub is ICandidateHub, System, IParamSubscriber {
   event paramChange(string key, bytes value);
 
   /*********************** init **************************/
-  function init() external onlyNotInit {
+  constructor(Registry registry, uint256 roundInterval_, uint256 validatorCount_) System(registry) {
     requiredMargin = INIT_REQUIRED_MARGIN;
     dues = INIT_DUES;
-    roundInterval = INIT_ROUND_INTERVAL;
-    validatorCount = INIT_VALIDATOR_COUNT;
+    roundInterval = roundInterval_;
+    validatorCount = validatorCount_;
     maxCommissionChange = MAX_COMMISSION_CHANGE;
     roundTag = 7;
-    alreadyInit = true;
   }
-  
+
+  function setControlRoundTimeTag(bool value) external onlyGov {
```

Comment:
This setter is for testing only and should never be called from production code. Since marked as onlyGov the additional risk is contained, still I would remove it from our code as soon as we can